aemanalyser-core/src/main/java/com/adobe/aem/analyser/AemAnalyserUtil.java

```diff
@@ -35,12 +39,33 @@ public class AemAnalyserUtil {
     }
 
     /** Used run modes */
-    static final List<String> USED_MODES = Arrays.asList("author", "author.dev", "author.stage",
-        "author.prod", "publish", "publish.dev", "publish.stage", "publish.prod");
+    static final List<String> AUTHOR_USED_MODES = Arrays.asList(
+        "author", "author.dev", "author.stage", "author.prod");
+    static final List<String> PUBLISH_USED_MODES = Arrays.asList(
+        "publish", "publish.dev", "publish.stage", "publish.prod");
+    static final List<String> ALL_USED_MODES = Stream.concat(
+        AUTHOR_USED_MODES.stream(), PUBLISH_USED_MODES.stream()).collect(Collectors.toList());
 
     /** Default runmode */
     private static final String DEFAULT_MODE = "(default)";
 
+    static List<String> getUsedModes(ServiceType[] runmodes) {
+        Set<ServiceType> rm = new HashSet<>(Arrays.asList(runmodes));
+
+        if (rm.size() == 0) {
```

Comment:
This looks like an error condition to me. It seems the code expects that the arrray exists and has at least one value.
Thinking of it, maybe an EnumSet is the better type than an array, as it ensures that a value is in there only once

Reply:
I made it throw an IllegalStateException if the size is 0, hope that works for you.